src/lib.rs

```diff
@@ -1748,12 +1748,10 @@ pub trait Itertools : Iterator {
     fn find_position<P>(&mut self, mut pred: P) -> Option<(usize, Self::Item)>
         where P: FnMut(&Self::Item) -> bool
     {
-        let mut index = 0usize;
-        for elt in self {
+        for (index, elt) in self.enumerate() {
             if pred(&elt) {
                 return Some((index, elt));
             }
-            index += 1;
```

Comment:
I saw places where the compiler generated different code for `enumerate` vs an explicit counter loop. I do not know whether one of them was strictly more performant.

I would accept this, unless there are objections. (@jswrenn - do you know something?)

Reply:
This could be written to call `find` instead, right?  That might be a good way to avoid the concern, since `find` will then use internal iteration too, which should be just as good as the manual `index`, if not better.

Something like this, maybe:
```rust
self.enumerate().find(|x| pred(&x.1))
```

Or there's also the way that's more like how `position` works,
```rust
self.try_fold(0, |i, x| {
    if pred(&x) { ControlFlow::Break((i, x)) }
    else { ControlFlow::Continue(i + 1) }
}).break_value()
```
but I don't know if your MSRV can do that one -- it might not be better, either.